ouroboros-consensus/src/Ouroboros/Consensus/BlockchainTime.hs

```diff
@@ -153,36 +154,64 @@ data TestClock =
   | Running !SlotNo
   deriving (Eq, Generic, NoUnexpectedThunks)
 
+-- | Options related to the ticking functionality of the 'TestBlockchainTime'.
+data TestTickOptions
+  = TimedTick
+    -- ^ Specifies that a thread should be spawned to automatically tick the
+    -- blockchain time on a periodic basis.
+      !NumSlots
+      -- ^ The maximum number of slots that the ticker thread should run for.
+      -- n.b. This is here to ensure that we terminate the thread (otherwise
+      -- the system will keep waiting).
+      !DiffTime
+      -- ^ The slot duration at which the blockchain time will be ticked.
+  | NoTimedTick
+    -- ^ The blockchain time will /not/ be automatically ticked on a
+    -- periodic basis.
+
 data TestBlockchainTime m = TestBlockchainTime
   { testBlockchainTime     :: BlockchainTime m
   , testBlockchainTimeDone :: m ()
     -- ^ Blocks until the end of the final requested slot.
+  , testBlockchainTimeTick :: m ()
+    -- ^ Manually tick the blockchain time.
   }
 
--- | Construct new blockchain time that ticks at the specified slot duration
+-- TODO @intricate: would it be preferred to just have two different functions
+-- rather than specifying options?
+--
+-- | Construct a new 'TestBlockchainTime'. The blockchain time can be manually
+-- ticked and, depending on the 'TestTickOptions' provided, it can also be
+-- configured to spawn a thread that automatically ticks at the specified slot
+-- duration.
 --
 -- NOTE: This is just one way to construct time. We can of course also connect
 -- this to the real time (if we are in IO), or indeed to a manual tick
 -- (in a demo).
 --
--- NOTE: The number of slots is only there to make sure we terminate the
--- thread (otherwise the system will keep waiting).
---
```

Comment:
This comment was moved to the documentation for the `TimedTick` constructor of the `TestTickOptions` type.